presto-main/src/main/java/com/facebook/presto/sql/planner/DependencyExtractor.java

```diff
@@ -44,11 +44,19 @@ private DependencyExtractor() {}
     public static List<Symbol> extractAll(Expression expression)
     {
         ImmutableList.Builder<Symbol> builder = ImmutableList.builder();
-        new Visitor().process(expression, builder);
+        new SymbolBuilderVisitor().process(expression, builder);
         return builder.build();
     }
 
-    private static class Visitor
+    // to extract qualified name with prefix
+    public static Set<QualifiedName> extract(Expression expression)
```

Comment:
Hmm.. this makes usage of this class more confusing (i.e., what's the difference between `extract` and `extractAll`). It may help if you call this method `extractNames`